packages/server-wallet/src/models/channel.ts

```diff
@@ -502,6 +501,29 @@ export class Channel extends Model implements ChannelColumns {
     return {targetBefore, targetAfter, targetTotal};
   }
 
+  /**
+   * The below logic assumes:
+   *  1. Each destination occurs at most once.
+   *  2. We only care about a single destination.
+   * One reason to drop (2), for instance, is to support ledger top-ups with as few state updates as possible.
+   */
+  allocationItemForParticipantIndex(index: number): AllocationItem | undefined {
+    if (index >= this.participants.length) {
+      throw new Error(
+        `${index} is greater than number of participants ${this.participants.length}`
+      );
+    }
+    const supported = this.supported;
+    if (!supported) {
+      throw new Error(`Channel has no supported state`);
+    }
+
+    const myDestination = this.participants[index].destination;
```

Comment:
I'll highlight that this code makes assumptions that we have multiple reasons to break:
1. In a unidirectional relationship, if the payer's `destination` is last in the list, then we can do special-case ledger top-ups for free.
2. In The Graph Network, we would like for a payment channel to serve multiple allocations. This will imply that a ledger channel will have multiple destinations for participant `1`.

Of course, a significant number of changes in the wallet are required to support either of these features.